web/client/plugins/Search.jsx

```diff
@@ -166,44 +166,99 @@ const SearchResultList = connect(selector, {
+ * a service, may have nested services, that allow you to search in several steps,
+ * </br> (e.g *search for a street and in the next step search for the street number.*)
+ * </br>The typical service with nested service needs to have some additional parameters, like **nestedPlaceholder**, **then**
+* @prop {string} cfg.searchOptions.services[].nestedPlaceholder the placeholder will be displayed in the input text, after you have performed the first search.
+* @prop {then[]} cfg.searchOptions.services[].then is the mandatory property to configure the nested service, every object in array,
+ * defines the steps of the search, performed by priority order
+
  * ```
  * {
- *     "type": "wfs",
- *     "filterTemplate": " AND SOMEPROP = '${properties.OLDPROP}'", // will be appended to the original filter, it gets the properties of the current selected item (of the parent service)
- *     "options": {
- *       ...
+ *  "nestedPlaceholder": "the placeholder will be displayed in the input text, after you have performed the first search",
+ *  "then": [{
+ *    "type": "wfs",
+ *    "priority": 1,
+ *    "displayName": "${properties.propToDisplay} ${properties.propToDisplay}",
+ *    "subTitle": " (a subtitle for the results coming from this service [ can contain expressions like ${properties.propForSubtitle}])",
+ *    "searchTextTemplate": "${properties.propToDisplay}",
+ *    "options": {
+ *      "staticFilter": " AND SOMEPROP = '${properties.OLDPROP}'", // will be appended to the original filter, it gets the properties of the current selected item (of the parent service)
+ *      "url": "/geoserver/wfs",
+ *      "typeName": "workspace:layer",
+ *      "queriableAttributes": ["attribute_to_query"],
+ *      "srsName": "EPSG:4326",
+ *       "maxFeatures": 10
  *     }
+ *  }]
  * }
+ *
  * ```
- * **note:** `searchTextTemplate` is useful to populate the search text input when a search result is selected, typically with "leaf" services.
- * @prop {array|boolean} cfg.withToggle when boolean, true uses a toggle to display the searchbar. When array, e.g  `["max-width: 768px", "min-width: 768px"]`, `max-width` and `min-width` are the limits where to show/hide the toggle (useful for mobile)
+ * **note:** `staticFilter` is valid for every service (even not nested), but if it is nested,
+ * you can use the selected feature from the parent service to specify this filter,
+ * that will be appended to the usual ilike filter used for searching text (wfs only)
+ * <br/>
+ * <br/>
+ * **note:** `searchTextTemplate` used to complete the text when an item is selected.
+ * (e.g. *I type "ro", I select an entry relative to "rome" and I want that the final text in search is "rome".*
+ * Can be used for every service that is a leaf (so it doesn't contain any nested service)
+* <br/> <br/> *Nested services can be used also with custom service, instead of `wfs`, in this case you will need to write your own code,
```

Comment:
Before, you should mention what custom services are. 
A custom service can be used to define your own remote service for text search.

to register a new service you can: 

```javascript
const {API} = require('../../MapStore2/web/client/api/searchText');

function myRoads(text, options) {
    axios.get(`/myService?text=${text}`).then(( results ) => { // results are [{title: "text", description: "description"}]
              return results.map((item) => ({
                        "type": "Feature",
                        "properties": {
                            "title": item.title,
                            "description": item.description
                        }
                    })
    })
}
API.Utils.setService("myRoads", myRoads);
```
The `myRoads` registered function must be a function that returns a `promise`.
The function gets as arguments the following parameters:
- `text`: the text to search
- `options`: the configurations of the service

And the promise returned must emit an array of geoJSON features. If the geometry is not present in your result you can define the `geomService` to retrive the geometry of the selected result. 

The `geomService` is a service like the others, that can use the `staticFilter` option as a template using the properties from the geojson result item. ( I notice that passing the item in option should be useful for custom geomServices that are not WFS, and less tricky that generating the staticFilter, maybe add a TODO :) ).